src/BCrypt/BCrypt.Helpers.cs

```diff
@@ -13,6 +13,11 @@ namespace PInvoke
     public static partial class BCrypt
     {
         /// <summary>
+        /// An array whose content doesn't matter.
+        /// </summary>
+        private static readonly byte[] OneElementDummyArray = new byte[1];
```

Comment:
After reading the code I understand, you want to use it as a marker and that's not possible with 0-sized arrays as the runtime can optimize all 0-size arrays to the same reference.